File: python/MASTER_SLAVE_WHOLE_PART_PROXY/MASTER_SLAVE_WHOLE_PART_PROXY/whole_part/Reservacion.py
from .DatosCliente import DatosCliente
from .DatosCompra  import DatosCompra
from .DatosFuncion import DatosFuncion
        
class Reservacion:
    
    datosCliente = DatosCliente()
    datosCompra  = DatosCompra()
    datosFuncion = DatosFuncion()
    
    def mostrarReservacion(self):
        print(
        f'''
        Reservación:
        
        Datos cliente:
            Nombres          : {self.datosCliente.nombres}
            Apellido paterno : {self.datosCliente.apellidoPaterno}
            Apellido materno : {self.datosCliente.apellidoMaterno}
            Edad             : {self.datosCliente.edad}
            Tipo de cliente  : {self.datosCliente.tipoCliente}
            
        Datos compra:
            Número de tarjeta               : {self.datosCompra.numeroTarjeta}
            Fecha de vencimiento de tarjeta : {self.datosCompra.fechaVencimiento}
            
        Datos función:
            Asiento  : {self.datosFuncion.asiento}
            Horario  : {self.datosFuncion.horario}
            pelicula : {self.datosFuncion.pelicula}
            
        '''
        )
        
    def devolverReservacion(self):
        return (self.datosCliente,
                self.datosCompra,
                self.datosFuncion)

File: php/ABSTRACT_FACTORY_DECORATOR/reservacion/Reservacion.php
<?php

require_once 'DatosCliente.php';
require_once 'DatosCompra.php';
require_once 'DatosFuncion.php';


class Reservacion {
    public $datosCliente;    
    public $datosCompra;
    public $datosFuncion;
    
    public function __construct($datosCliente, $datosCompra, $datosFuncion) {
        $this->datosCliente = $datosCliente;
        $this->datosCompra = $datosCompra;
        $this->datosFuncion = $datosFuncion;
    }
    
    public function mostrarReservacion() {
        $informacion = "<h2>Datos de Cliente</h2>";
        $informacion .= "Nombre: " .  $this->datosCliente->getNombre();
        $informacion .= "<br/>Apellido Paterno: " . $this->datosCliente->getApellidoPaterno();
        $informacion .= "<br/>Apellido Materno: " . $this->datosCliente->getApellidoMaterno();
        $informacion .= "<br/>Edad: " . $this->datosCliente->getEdad();
        $informacion .= "<br/>Tipo de cliente: " . $this->datosCliente->getTipoCliente() . '<br/>';
        $informacion .= "<h2>Datos de Funcion</h2>";
        $informacion .= "<br/>Pelicula: " . $this->datosFuncion->getPelicula();
        $informacion .= "<br/>Fecha: " . $this->datosFuncion->getDia() .
                "/".$this->datosFuncion->getMes() ."/". $this->datosFuncion->getAnio();
        $informacion .= "<br/>Hora: " . $this->datosFuncion->getHora();
        $informacion .= "<br/>Asiento: " . $this->datosFuncion->getAsiento() . '<br/>';
        $informacion .= "<h2>Datos de Compra</h2>";
        $informacion .= "<br/>Número de tarjeta: " . $this->datosCompra->getNumeroTarjeta();
        $informacion .= "<br/>Fecha: " . date('Y-m-d' , $this->datosCompra->getFechaVencimiento()). "</br>";
        
        return $informacion;
    }


    
}
